components/infobox/wikis/starcraft/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,510 @@
+		end
+
+		yearsActive = CustomPlayer._getYearsActive(years)
+
+		yearsActive = string.gsub(yearsActive, '<br>', '', 1)
+
+		if String.isNotEmpty(category) and String.isNotEmpty(yearsActive) then
+			yearsActive = yearsActive .. '[[Category:' .. category .. ']]'
+		end
+
+		CustomPlayer._setVarsForVS(vs)
+	end
+
+	return yearsActive
+end
+
+function CustomPlayer._getYearsActive(years)
+	local yearsActive = ''
+	local tempYear = nil
+	local firstYear = true
+
+	for i = _START_YEAR, _CURRENT_YEAR do
+		if years[i] then
+			if (not tempYear) and (i ~= _CURRENT_YEAR) then
+				if firstYear then
+					firstYear = nil
+				else
+					yearsActive = yearsActive .. '<br/>'
+				end
+				yearsActive = yearsActive .. years[i]
+				tempYear = years[i]
+			end
+			if i == _CURRENT_YEAR then
+				if tempYear then
+					yearsActive = yearsActive .. '&nbsp;-&nbsp;<b>Present</b>'
+				else
+					yearsActive = yearsActive .. '<br/><b>Present</b>'
+				end
+			elseif not years[i + 1] then
+				if tempYear ~= years[i] then
+					yearsActive = yearsActive .. '&nbsp;-&nbsp;' .. years[i]
+				end
+				tempYear = nil
+			end
+		end
+	end
+
+	return yearsActive
+end
+
+function CustomPlayer._setVarsForVS(table)
+	for key1, item1 in pairs(table) do
+		for key2, item2 in pairs(item1) do
+			for key3, item3 in pairs(item2) do
+				Variables.varDefine(key1 .. '_vs_' .. key2 .. '_' .. key3, item3)
+			end
+		end
+	end
+end
+
+function CustomPlayer._addScoresToVS(vs, opponents, player)
+	local plIndex = 1
+	local vsIndex = 2
+	--catch matches vs empty opponents
+	if opponents[1] and opponents[2] then
+		if opponents[2].name == player then
+			plIndex = 2
+			vsIndex = 1
+		end
+		local plOpp = opponents[plIndex]
+		local vsOpp = opponents[vsIndex]
+
+		local prace = CleanRace[plOpp.match2players[1].extradata.faction] or 'r'
+		local orace = CleanRace[vsOpp.match2players[1].extradata.faction] or 'r'
+
+		vs[prace][orace].win = vs[prace][orace].win + (tonumber(plOpp.score or 0) or 0)
+		vs[prace][orace].loss = vs[prace][orace].loss + (tonumber(vsOpp.score or 0) or 0)
+
+		vs['total'][orace].win = vs['total'][orace].win + (tonumber(plOpp.score or 0) or 0)
+		vs['total'][orace].loss = vs['total'][orace].loss + (tonumber(vsOpp.score or 0) or 0)
+
+		vs[prace]['total'].win = vs[prace]['total'].win + (tonumber(plOpp.score or 0) or 0)
+		vs[prace]['total'].loss = vs[prace]['total'].loss + (tonumber(vsOpp.score or 0) or 0)
+
+		vs['total']['total'].win = vs['total']['total'].win + (tonumber(plOpp.score or 0) or 0)
+		vs['total']['total'].loss = vs['total']['total'].loss + (tonumber(vsOpp.score or 0) or 0)
+	end
+
+	return vs
+end
+]==]
+
+function CustomPlayer:calculateEarnings()
+	local earningsTotal
+	earningsTotal, _earningsGlobal = CustomPlayer._getEarningsMedalsData(_player.pagename)
+	earningsTotal = Math.round{earningsTotal}
+	return earningsTotal, _earningsGlobal
+end
+
+function CustomPlayer._getEarningsMedalsData(player)
```

Comment:
Should we move Medals to an independent commons module?

Reply:
i think only sc and sc2 use it

Reply:
R6 previously had an independent module that calculated and showed it.

Reply:
doing it independent of the earnings calculation would basically increase the lpdb query runtime ~1/3 of the runtime it would have if done together which is why i oped to do them together in the first place